tools/c7n_guardian/c7n_guardian/cli.py

```diff
@@ -168,15 +220,29 @@ def disable(config, tags, accounts, master, debug,
 
 
 def get_session(role, session_name, profile, region):
+    # TODO appears not to be threadsafe .... though I'm creating a client each time ...
+    # Getting like 2020-05-18 19:15:28,430: c7n-guardian:ERROR
```

Comment:
are you on osx running? what version of python are you using? this isn't a thread safety to be clear, its a python fork issue with multiprocessing, ie i suspect your getting https://bugs.python.org/issue33725 which is resolved in python 3.8

Reply:
> Thanks for the pull request! If you don't mind could you sign the CLA

Sure thing

Reply:
> are you running on osx? what version of python are you using? this isn't a thread safety to be clear, its a python fork issue with multiprocessing, ie i suspect your getting https://bugs.python.org/issue33725 which is resolved in python 3.8

I think you're right - it's fixed now in the latest version of the PR

Reply:
Thanks for timely review btw Kapil ... and apologies for the mass of sweeping changes, hope you can follow the intentions & implementations ok

Reply:
tbh black here doesn't help on total diff when evaluating concurrency considerations as its mostly line noise, but there was enough git lineage to piece things some bits together,  tbd on the extraneous formatting line noise on the overall diff.